realm/realm-library/src/main/cpp/io_realm_internal_objectstore_OsMongoCollection.cpp

```diff
@@ -100,14 +117,14 @@ JNIEXPORT void JNICALL
 Java_io_realm_internal_objectstore_OsMongoCollection_nativeFindOne(JNIEnv* env,
```

Comment:
A bit hard to say with the lack of overview here, but it seems like many of these native methods are doing almost the same. Maybe it would be worth just differentiating them by a _selector_ or similar as in ex. https://github.com/realm/realm-java/blob/v10/realm/realm-library/src/main/cpp/io_realm_ApiKeyAuth.cpp#L79

Reply:
I thought of that too, there is a lot of code repetition. I will definitely address this.